models/port.go

```diff
@@ -4,9 +4,10 @@ import core_v1 "k8s.io/api/core/v1"
 
 type Ports []Port
 type Port struct {
-	Name     string `json:"name"`
-	Protocol string `json:"protocol"`
-	Port     int32  `json:"port"`
+	Name        string `json:"name"`
+	Protocol    string `json:"protocol"`
+	AppProtocol string `json:"appProtocol"`
```

Comment:
You might want to add here the `omitempty` tag. So if it is nil, the field won't be part of the json. 
In the front-end you already have the field as optional `?` so shouldn't be a problem.

Reply:
agree